### PR TITLE
Added data folder as volume for backend-docker-start

### PR DESCRIPTION
**Pull request by ichim-david**
This way we make it easy for the beginners that have the docs to use backend-docker-start to run Plone backend to have their data persisted

### CODE REVIEW

**Comment by wesleybl** [comment]
@ichim-david I tested but couldn't log in. I tried:

admin:admin
admin:secret

**Comment by ichim-david** [comment]
> @ichim-david I tested but couldn't log in. I tried:
> 
> admin:admin admin:secret
@wesleybl 

> @ichim-david I tested but couldn't log in. I tried:
> 
> admin:admin admin:secret

@wesleybl works for me with admin:admin, removed my data and tried again and same result. try in incognito

**Comment by wesleybl** [comment]
> @wesleybl works for me with admin:admin, removed my data and tried again and same result. try in incognito

I tried again and it worked.

The data was percised.

**Review by wesleybl — COMMENTED** [review]
I tested it and it worked. Just one doubt. Won't this make searches in the `package/volto` folder less time consuming? We will have many files in the data folder. Maybe we should use the default volume path?

**Comment by ichim-david** [comment]
> I tested it and it worked. Just one doubt. Won't this make searches in the `package/volto` folder more time consuming? We will have many files in the data folder. Maybe we should use the default volume path?

@wesleybl could be an issue depending on the editor. For VS Code it respects the .gitignore settings if the "Gear icon Use Exclude settings and ignore Files" option is on.
Similar settings is found for Pycharm.
If this is a problem try another way to define the volume and come up with a suggestion for a change, at least that way we have a comparison on what might work best given each pro and cons.

**Comment by wesleybl** [comment]
@ichim-david 

```bash
docker run -it --rm --name=backend -p 8080:8080 -v volto-backend-data:/data -e SITE=Plone -e ADDONS='$(KGS)' $(DOCKER_IMAGE)
```

This way we won't have extra files in the repository.

**Comment by ichim-david** [comment]
@wesleybl added your suggestion in https://github.com/plone/volto/pull/6168/commits/6f4b382b15b5ca178b0d0191a5becf9cd7284635

**Review by sneridagh — CHANGES_REQUESTED** [review]
@ichim-david we always refrained to add persistant in there to make it harder for people to be tempted and use that in "production".
Not that I'm opposed to add it, but I'd drop some lines of docs for it, including the way of wipe that persistence, if needed.

**Comment by wesleybl** [comment]
> including the way of wipe that persistence, if needed.

@sneridagh maybe have a new make command, or is it not necessary? The command to clean would be:

```
docker volume rm volto-backend-data
```

File: packages/volto/news/6157.bugfix
Comment (stevepiercy) [inline packages/volto/news/6157.bugfix:1]:
```suggestion
Persist data for the `backend-docker-start` Docker container in a Docker volume named `volto-backend-data`.
```

File: packages/volto/news/6157.bugfix
Comment (stevepiercy) [inline packages/volto/news/6157.bugfix:2]:
```suggestion
This way the data is persisted between runs of the container.
You can also delete the `data` volume to start fresh.
```

File: docs/source/contributing/developing-core.md
Comment (stevepiercy) [inline docs/source/contributing/developing-core.md:178]:
```suggestion

When you run this command for the first time, it will download Docker images, configure the backend, create a Docker volume to persist the data named `volto-backend-data`, and start the backend.
You can also delete the volume, and recreate it to start fresh, with the same command.
```

File: docs/source/contributing/developing-core.md
Comment (stevepiercy) [inline docs/source/contributing/developing-core.md:179]:
```suggestion
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
I will need to review it again, but I think these suggestions remove some redundancies with other sentences in the same page.

File: docs/source/contributing/developing-core.md
Comment (ichim-david) [reply docs/source/contributing/developing-core.md:177]: @stevepiercy I committed the other changes except this one since I needed to add the info on how to delete the volume check out the notes below. It took so long for me to get the docker volume command to show up as a copy target and the text that follows it as part of the note :( 
It cannot receive any extra ticks to mark it as shell as it messes up the text below it.

**Review by stevepiercy — APPROVED** [review]
I moved the admonition just below the heading because it disrupted the flow.

I also used nested MyST blocks to put a code block inside the admonition. For every level of nesting, add a backtick.

I think this is good to merge, but you should give it one final pass.

**Review by stevepiercy — APPROVED** [review]
@sneridagh could I get a review from you, too, please? I think this addresses your concerns. See preview at https://volto--6168.org.readthedocs.build/contributing/developing-core.html#start-the-backend-and-volto

**Review by wesleybl — APPROVED** [review]
LGTM

**Comment by stevepiercy** [comment]
@ichim-david OK to merge, after my revisions?

**Comment by ichim-david** [comment]
@stevepiercy yup from my point of view it's ready to merge as well

**Pull request merged**
